packages/block-editor/src/components/block-list/index.js

```diff
@@ -36,6 +37,84 @@ const forceSyncUpdates = ( WrappedComponent ) => ( props ) => {
 		</AsyncModeProvider>
 	);
 };
+
+const BlockListItemWrapper = ( { blockClientIds, isBlockInSelection, ...props } ) => {
+	const ref = useRef( null );
+	const [ resetAnimation, updateReset ] = useState( false );
+	const [ transform, setTransform ] = useState( { x: 0, y: 0 } );
+	const previous = ref.current ? ref.current.getBoundingClientRect() : null;
+	const prefersReducedMotion = useReducedMotion();
+	useLayoutEffect( () => {
+		if ( prefersReducedMotion ) {
+			return;
+		}
+		ref.current.style.transform = 'none';
+		const destination = ref.current.getBoundingClientRect();
+		const newTransform = {
+			x: previous ? previous.left - destination.left : 0,
+			y: previous ? previous.top - destination.top : 0,
+		};
+		ref.current.style.transform = `translate3d(${ newTransform.x }px,${ newTransform.y }px,0)`;
+		updateReset( true );
+		setTransform( newTransform );
+	}, [ blockClientIds ] );
+	useEffect( () => {
+		if ( resetAnimation ) {
+			updateReset( false );
+		}
+	}, [ resetAnimation ] );
+	const animationProps = useSpring( {
+		from: {
+			...transform,
+		//	opacity: 0,
+		//	scale: 0,
+		},
+		to: {
+			x: 0,
+			y: 0,
+		//	scale: isBlockInSelection ? 1 : 0,
+		//	opacity: isBlockInSelection ? 0 : 1,
+		},
+		reset: resetAnimation,
+		config: { mass: 5, tension: 2000, friction: 200 },
+		immediate: prefersReducedMotion,
+	} );
+
+	return (
+		<animated.div
+			ref={ ref }
+			className={ classnames( 'editor-block-list__block-animated-container', {
```

Comment:
This adds a new container for each block. We could probably consider whether it's possible to reuse the same block container.